### DISCUSSION TITLE
v0.1.0 errors SensorError(BadChip)

### DISCUSSION THREAD

**Discussion by seam345** [jessebraham/icm42670#4] [General]
After updating the example https://github.com/jessebraham/icm42670-examples/pull/1 when checking out v0.1.0 I found it would just panic at `let mut icm = Icm42670::new(i2c, Address::Primary).unwrap();` putting it in a match statement and printing out the error I get `SensorError(BadChip)`. Still very new to this I am very unsure what changed, or if I have done something wrong, but v0.1.0 is the only version on crates.io maybe a new release is needed as Main worked for me.

**Comment by jessebraham** [comment]
Sorry about that, I have actually fixed this issue in [this commit](https://github.com/jessebraham/icm42670/commit/eb0483bbb0ec485b35c8d38c1f36a02bcc899e0a) but never created a new release. I'll try to publish a patch release including this soon.

The reason this happens is that (for whatever reason) there is the ICM-42670, and then also the ICM-42607 (which is the mass-production version, and differs __only__ by device ID).

**Reply by seam345** [reply]
Thanks for linking the commit, nice to better understand how it all works!

**Comment by jessebraham** [comment]
I've published `0.1.1` which should hopefully resolve this issue.

**Comment by seam345** [comment]
Yep updated my cargo.toml and all just works, thank you for being so fast thought i was gonna have to grab it from git for a while :)

**Comment by jessebraham** [comment]
Not a problem, glad it worked for you!
